File: core/src/net/brenig/pixelescape/game/worldgen/TerrainPair.java
package net.brenig.pixelescape.game.worldgen;

import net.brenig.pixelescape.PixelEscape;
import net.brenig.pixelescape.game.World;
import net.brenig.pixelescape.lib.Reference;

/**
 * Used to store terrain
 */
public class TerrainPair {

	private int bot;
	private int top;

	public TerrainPair(int bot, int top) {
		this.setBot(bot);
		this.setTop(top);
	}

	public int getBot() {
		return bot;
	}

	public void setBot(int bot) {
		this.bot = bot;
	}

	public void setTop(int top) {
		this.top = top;
	}

	public int getTop() {
		return top;
	}

	public void render(PixelEscape game, World world, float x, float y, float yTranslation, float delta) {
		game.getRenderManager().begin();
		//Draw Bottom (y=0) blocks
		game.getRenderManager().rect(x, y, Reference.BLOCK_WIDTH, getBot() * Reference.BLOCK_WIDTH + yTranslation);
		//Draw Top (y=worldHeight) blocks
		game.getRenderManager().rect(x, y + world.getWorldHeight(), Reference.BLOCK_WIDTH, (getTop() * Reference.BLOCK_WIDTH - yTranslation) * -1);

	}
}

File: core/src/net/brenig/pixelescape/game/worldgen/TerrainPair.kt
package net.brenig.pixelescape.game.worldgen

import net.brenig.pixelescape.PixelEscape
import net.brenig.pixelescape.game.World
import net.brenig.pixelescape.game.data.constants.Reference

/**
 * Used to store terrain
 */
class TerrainPair(bot: Int, top: Int) {

    var bot: Int = 0
    var top: Int = 0

    init {
        this.bot = bot
        this.top = top
    }

    fun render(game: PixelEscape, world: World, x: Float, y: Float, yTranslation: Float, delta: Float) {
        game.renderManager.begin()
        //Draw Bottom (y=0) blocks
        game.renderManager.rect(x, y, Reference.BLOCK_WIDTH.toFloat(), bot * Reference.BLOCK_WIDTH + yTranslation)
        //Draw Top (y=worldHeight) blocks
        game.renderManager.rect(x, y + world.worldHeight, Reference.BLOCK_WIDTH.toFloat(), (top * Reference.BLOCK_WIDTH - yTranslation) * -1)

    }
}
